apps/src/dance/lab2/ProgramExecutor.ts

```diff
@@ -100,9 +100,10 @@ export default class ProgramExecutor {
   }
 
   /**
-   * Preview the program. Compiles student code and calls on the native API to draw the first frame.
+   * Preview the program. Compiles student code and calls on the native API to draw a frame.
    */
   async preview() {
+    this.nativeAPI.setForegroundEffectsInPreviewMode(true);
```

Comment:
@sanchitmalhotra126 I see we have some other functions here like here like `reset()` that manage most of the specifics of calling the dance party API. Not sure if you'd rather this go in something like that (`setup()`?)

Reply:
I think keeping it here is fine since it's preview-related - `reset()` is more or less analogous to `reset` in Dance.js in that it handles stopping drawing and playback whenever the Reset button is clicked.

Reply:
Will we need to add a `setTimeout` here when we migrate to lab2 and use this file for both the modal and vis column?

Reply:
Not sure what you mean, can you elaborate?

Reply:
Currently, `ProgramExecutor` is only being used for the `AiVisualizationPreview` that @sanchitmalhotra126 implemented. But when we move to lab2, I think we'll be solely using `ProgramExecutor` and not `Dance.js`. At that point, I am wondering if we'll have the same issue with the preview when reset is called.

Reply:
I don't think we'll have a problem b/c I added the setTimeout fix for preview here as well?

Reply:
Oh yeah! 🤦‍♀️